src/components/CourseDetail/CourseDetail.js

```diff
@@ -1,28 +1,43 @@
 import React from 'react';
 import './CourseDetail.scss';
-import { useSelector } from 'react-redux';
+import { useSelector, useDispatch } from 'react-redux';
 import YouTube from 'react-youtube';
+import { completeCourse, teacherSignIn } from '../../utils/apiCalls';
 
 const CourseDetail = ({ id }) => {
-  const { courses } = useSelector(state => state);
+  const dispatch = useDispatch();
+  const { courses, teacher } = useSelector(state => state);
```

Comment:
This could be a place to refactor in future iterations, but well done making it work.